src/main.js

```diff
@@ -4,89 +4,94 @@
 
 const $ = (selector) => document.querySelector(selector);
 
-const $titleInput = $('#titleInput');
-const $authorInput = $('#authorInput');
 const $addBookBtn = $('#addBookBtn');
 const $bookContainer = $('#book-container');
-
-let bookCollection = [];
+const $titleInput = $('#titleInput');
+const $authorInput = $('#authorInput');
 
 // ***************
 // Local Storage
 // ***************
 
-function saveLocal() {
-  const catchCollection = JSON.stringify(bookCollection);
-  localStorage.setItem('bookCollection', catchCollection);
-}
+class BookApp {
+  constructor() {
+    this.bookCollection = [];
+  }
 
-function getLocal() {
-  if (localStorage.getItem('bookCollection')) {
-    bookCollection = JSON.parse(localStorage.getItem('bookCollection'));
+  book = (titlePar, authorPar) => {
+    const bookObj = {
+      title: titlePar,
+      author: authorPar,
+    };
+    return bookObj;
   }
-}
 
-// ***************
-// main functions
-// ***************
+  saveLocal = () => {
+    const catchCollection = JSON.stringify(this.bookCollection);
+    localStorage.setItem('bookCollection', catchCollection);
+  }
 
-function bookTemplate(el) {
-  return `
-<h3 class="bookTitle">${el.title}</h3>
-<p class="bookAuthor">${el.author}</p>
-<button type='button' class="removeBookBtn">Remove</button>
-`;
-}
+  getLocal = () => {
+    if (localStorage.getItem('bookCollection')) {
+      this.bookCollection = JSON.parse(localStorage.getItem('bookCollection'));
+    }
+  }
 
-function renderBooks() {
-  $bookContainer.innerHTML = '';
-  bookCollection.forEach((el) => {
-    const article = document.createElement('article');
-    article.className = 'article-book';
-    article.innerHTML = bookTemplate(el);
-    $bookContainer.appendChild(article);
-  });
-}
+  // ***************
+  // main functions
+  // ***************
 
-function createRemoveBtn() {
-  const $removeBookBtn = document.querySelectorAll('.removeBookBtn');
-  $removeBookBtn.forEach((el, index) => {
-    el.addEventListener('click', () => {
-      bookCollection.splice(index, 1);
-      renderBooks();
-      createRemoveBtn();
-      saveLocal();
+  renderBooks = () => {
+    $bookContainer.innerHTML = '';
+    this.bookCollection.forEach((el) => {
+      const article = document.createElement('article');
+      article.className = 'article-book';
+      article.innerHTML = `
+  <h3 class="bookTitle">"${el.title}" by ${el.author}</h3>
+  <button type='button' class="removeBookBtn">Remove</button>
+  `;
+      $bookContainer.appendChild(article);
     });
-  });
-}
+  }
 
-function displayBookCollection() {
-  $bookContainer.innerHTML = '';
-  renderBooks();
-  createRemoveBtn();
-}
+  createRemoveBtn = () => {
+    const $removeBookBtn = document.querySelectorAll('.removeBookBtn');
+    $removeBookBtn.forEach((el, index) => {
+      el.addEventListener('click', () => {
+        this.bookCollection.splice(index, 1);
```

Comment:
- [ ] OPTIONAL
It will be a good practice to favor the use of array methods such as `slice` over `splice`, which doesn't mutate the array.